spring-web/src/main/java/org/springframework/http/codec/json/AbstractJackson2Decoder.java

```diff
@@ -217,6 +244,32 @@ private ObjectReader getObjectReader(
 				mapper.readerFor(javaType);
 	}
 
+	/**
+	 * Provides the ability for subclasses to customize the {@link ObjectReader} for deserialization from a stream.
+	 * @param reader the {@link ObjectReader} available for customization
+	 * @param mimeType the MIME type associated with the input stream
+	 * @param elementType the expected type of elements in the output stream
+	 * @param hints additional information about how to do encode
+	 * @return the customized {@link ObjectReader}
+	 */
+	protected Mono<ObjectReader> customizeReaderFromStream(@NonNull ObjectReader reader, @Nullable MimeType mimeType,
+			ResolvableType elementType, @Nullable Map<String, Object> hints) {
+		return Mono.just(customizeReader(reader, mimeType, elementType, hints));
+	}
```

Comment:
This was done intentionally so that if someone extending the class wanted to customize leveraging data in the Reactor Context they could.  They would be able to do something like this:

```
@Override
protected Mono<ObjectReader> customizeReaderFromStream(@NonNull ObjectReader reader, @Nullable MimeType mimeType, ResolvableType elementType, @Nullable Map<String, Object> hints) {
	return Mono.deferContextual(ctx -> {
	    if(ctx.getOrDefault("something", false) {
	        // customize reader
	    }
	    return reader;
	});
}
```

I have an actual use case for this.

Reply:
Apologies for the delay. I'm still not too keen on a `Mono` and a `flatMap` for a local method. To provide access to the Reactor Context, I would prefer if the top-level method accessed the `ContextView` and provided to the protected customization methods. For example: 
```java
return Flux.deferContextual(context -> {
	ObjectReader reader = getObjectReader(mapper, elementType, hints);
	customizeReader(reader, mimeType, elementType, hints, context);
	// ...
});
```

Reply:
I can make those changes, if you're okay with those.

Reply:
What you suggested makes total sense. If you are comfortable making the change, I'm good with it.